src/progress/pct/v11/dynrun.p

```diff
@@ -0,0 +1,136 @@
+USING Progress.Json.ObjectModel.JsonArray.
+USING Progress.Json.ObjectModel.JsonObject.
+USING Progress.Json.ObjectModel.ObjectModelParser.
+
+ROUTINE-LEVEL ON ERROR UNDO, THROW.
+
+DEFINE NEW SHARED VARIABLE pctVerbose AS LOGICAL NO-UNDO.
+DEFINE VARIABLE noErrorOnQuit AS LOGICAL NO-UNDO.
+
+DEFINE VARIABLE i AS INTEGER NO-UNDO INITIAL ?.
+
+DEFINE TEMP-TABLE ttParams NO-UNDO
+  FIELD key AS CHARACTER
+  FIELD val AS CHARACTER.
+
+FUNCTION getParameter RETURNS CHARACTER (k AS CHARACTER).
+  FIND ttParams WHERE ttParams.key EQ k NO-LOCK NO-ERROR.
+  RETURN (IF AVAILABLE ttParams THEN ttParams.val ELSE ?).
+END FUNCTION.
+
+DEFINE VARIABLE jsonParser AS CLASS ObjectModelParser NO-UNDO.
+DEFINE VARIABLE configJson AS CLASS JsonObject NO-UNDO.
+
+DEFINE VARIABLE ppEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE dbEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE dbEntry   AS CLASS JsonObject NO-UNDO.
+DEFINE VARIABLE prmEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE prmEntry   AS CLASS JsonObject NO-UNDO.
+DEFINE VARIABLE outprmEntries AS CLASS JsonArray NO-UNDO.
+DEFINE VARIABLE zz AS INTEGER     NO-UNDO.
+DEFINE VARIABLE xx AS INTEGER     NO-UNDO.
+DEFINE VARIABLE out1 AS CHARACTER   NO-UNDO.
+DEFINE VARIABLE out2 AS CHARACTER   NO-UNDO.
+
+ASSIGN jsonParser = NEW ObjectModelParser().
+ASSIGN configJson = CAST(jsonParser:ParseFile(SESSION:PARAMETER), JsonObject).
+
+ASSIGN pctVerbose = configJson:getLogical("verbose").
+
+// DB connections + aliases
+ASSIGN dbEntries = configJson:GetJsonArray("databases").
+DO zz = 1 TO dbEntries:Length:
+  ASSIGN dbEntry = dbEntries:GetJsonObject(zz).
+  IF pctVerbose THEN MESSAGE "Connecting to '" + dbEntry:getCharacter("connect") + "'".
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Statement MESSAGE should be on a separate line [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.OneStatementPerLine)